generation/utils.py

```diff
@@ -0,0 +1,48 @@
+from datetime import timedelta
+from flask import current_app
+
+from flexmeasures.data.models.generic_assets import GenericAsset, GenericAssetType
+from flexmeasures.data.models.time_series import Sensor
+from flexmeasures.data.config import db
+from timely_beliefs.sensors.func_store import knowledge_horizons
+
+from .. import DEFAULT_COUNTRY_CODE  # noqa: E402
+
+
+
+def ensure_generation_sensors():
+    """
+    Ensure a GenericAsset exists to model the transmission zone for which this plugin gathers
+    generation data, plus sensors for relevant data we collect.
+    """
+    country_code = current_app.config.get("ENTSOE_COUNTRY_CODE", DEFAULT_COUNTRY_CODE)
+    transmission_zone_type = GenericAssetType.query.filter(
+        GenericAssetType.name == "transmission zone"
+    ).one_or_none()
+    if not transmission_zone_type:
+        current_app.logger.info("Adding transmission zone type ...")
+        transmission_zone_type = GenericAssetType(name="transmission zone", description="A grid regulated & balanced as a whole, usually a national grid.")
+        db.session.add(transmission_zone_type)
+    ga_name = f"{country_code} transmission zone"
+    transmission_zone = GenericAsset.query.filter(GenericAsset.name == ga_name).one_or_none()
+    if not transmission_zone:
+        current_app.logger.info(f"Adding {ga_name} ...")
+        transmission_zone = GenericAsset(
+            name=ga_name,
+            generic_asset_type=transmission_zone_type,
+            account_id=None  # public
+        )
+    for sensor_name, unit in (("Overall generation", "MWh"), ("Solar", "MWh"), ("Onshore wind", "MWh"), ("Offshore wind", "MWh"), ("CO2", "kg/h")):
```

Comment:
1. I'd rather having all sensors use units of flow, i.e. MW and kg/h.
2. Given that the emission sensor here will measure an entire transmission zone, I'd suggest using at least t/h (t stands for tonne, a.k.a. metric ton), maybe even kt/h (~6 kt/h is a reasonable value for NL, ~50 kt/h for DE).

Reply:
Actually, we should approach the CO₂ eq. sensor as a ratio, similar to a unit price. Then the unit would become kg/MWh
(analogous to EUR/MWh). I think that makes more sense. For example, when plotting the CO₂ eq. sensor, a time series plot of kg/MWh tells you immediately how much emissions you would cause by consuming 1 MWh. In contrast, a time series plot of t/h tells you how much emissions were caused in that hour for the entire transmission zone, but doesn't tell you how your consumption relates to that.